modules/storage/storage-account/blobServices/containers/deploy.bicep

```diff
@@ -0,0 +1,73 @@
+@maxLength(24)
+@description('Required. Name of the Storage Account.')
+param storageAccountName string
+
+@description('Optional. Name of the blob service.')
+param blobServicesName string = 'default'
+
+@description('Required. The name of the storage container to deploy')
+param name string
+
+@description('Optional. Name of the immutable policy.')
+param immutabilityPolicyName string = 'default'
+
+@allowed([
+  'Container'
+  'Blob'
+  'None'
+])
+@description('Optional. Specifies whether data in the container may be accessed publicly and the level of access.')
+param publicAccess string = 'None'
+
+@description('Optional. Configure immutability policy.')
+param immutabilityPolicyProperties object = {}
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'')
+param roleAssignments array = []
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-06-01' existing = {
+  name: storageAccountName
+
+  resource blobServices 'blobServices@2021-06-01' existing = {
+    name: blobServicesName
+  }
+}
+
+resource container 'Microsoft.Storage/storageAccounts/blobServices/containers@2019-06-01' = {
+  name: name
+  parent: storageAccount::blobServices
+  properties: {
+    publicAccess: publicAccess
+  }
+}
+
+module immutabilityPolicy 'immutabilityPolicies/deploy.bicep' = if (!empty(immutabilityPolicyProperties)) {
+  name: immutabilityPolicyName
+  params: {
+    storageAccountName: storageAccount.name
+    blobServicesName: storageAccount::blobServices.name
+    containerName: container.name
+    immutabilityPeriodSinceCreationInDays: contains(immutabilityPolicyProperties, 'immutabilityPeriodSinceCreationInDays') ? immutabilityPolicyProperties.immutabilityPeriodSinceCreationInDays : 365
```

Comment:
Now that is very interesting @bmoore-msft . Does it work like that, i.e. will it use the module's default?
I'm asking because I am quite certain I received errors when doing that. Especially in cases where there is an array of allowed values (and null isn't in it). I will re-test because that would be a big improvement for us - and make maintenance a lot easier.

Reply:
Alright, just did with the following setup:

`submodule.bicep`:
```bicep
param testPar1 string = 'par1'

param testPar2 bool = true

param testPar3 object = {
  propertyA: 'par3a'
  propertyB: 'par3b'
}

output oTestPar1 string = testPar1
output oTestPar2 bool = testPar2
output oTestPar3 object = testPar3
```

`topModule.bicep`:
```bicep
module testDeployment './submodule.bicep' = {
  name: 'testMe'
  params: {
    testPar1: true == false ? 'value' : any(null)
    testPar2: true == false ? true : any(null)
    testPar3: true == false ? {
      par: 'par'
    } : any(null)
  }
}

output oTestPar1 string = testDeployment.outputs.oTestPar1
output oTestPar2 bool = testDeployment.outputs.oTestPar2
output oTestPar3 object = testDeployment.outputs.oTestPar3
```

Tested it with the booleans both `true` & `false`. These are the results:
- If I don't specify `any(null)`, Bicep argues for example `The property "testPar2" expected a value of type "bool" but the provided value is of type "bool | null".bicep(BCP036)`. Hence I added it for all 3 parameters
- If all 3 values are true, the module returns the values I put in (so far, so expected)
- If I now set the condition to `false`, and hence pass the `any(null)` into the references module, Bicep failes with the error: `Error: Code=InvalidDeploymentParameterValue; Message=The value of deployment parameter 'testPar1' is null. Please specify the value or use the parameter reference. See https://aka.ms/resource-manager-parameter-files for details.`

So unless I'm missing something, it seems it's still the case that we have to duplicate default values and cannot pass `null` to modules (to fallback to its default).